core/src/main/java/com/amplifyframework/util/UserAgent.java

```diff
@@ -92,39 +118,61 @@ public static synchronized void reset() {
 
     /**
      * Gets a String to use as the value of a User-Agent header.
+     * <p>
+     * Clients should use this method only to get the user agent portion generated by library.
+     * and use {@link #configure(Map)} to set the actual user agent metadata passed to the SDK.
+     * <br/>
+     * This string does not contain extra metadata added by the SDK.
+     * </p>
      * Uses default Android user-agent if {@link #configure(Map)} has not been called yet.
+     *
      * @return A value for a User-Agent header.
      */
     @SuppressLint("SyntheticAccessor")
     @NonNull
     public static String string() {
         if (instance == null) {
             LOG.debug("User-Agent is not yet configured. Returning default Android user-agent.");
-            return forAndroid();
+            return new UserAgent(
+                    Platform.ANDROID.getLibraryName(),
+                    BuildConfig.VERSION_NAME,
+                    new HashMap<>()
+            ).toString();
         }
 
-        return instance;
+        return instance.toString();
     }
 
     /**
-     * Return user-agent string of the form:
-     * amplify-android:x.x.x md/xxx/yyy-zzz /md/locale/en_US
-     * systemName and systemVersion not required, as it is part of sdk user-agent string.
-     * @return user-agent string
+     * Gets the string representation of the User-agent.
+     *
+     * @return A sting value for user-agent
      */
-    private static String forAndroid() {
-        return new UserAgent.Builder()
-                .libraryName(Platform.ANDROID.getLibraryName())
-                .libraryVersion(BuildConfig.VERSION_NAME)
-                .deviceManufacturer(Build.MANUFACTURER)
-                .deviceName(Build.MODEL)
-                .userLanguage(System.getProperty("user.language"))
-                .userRegion(System.getProperty("user.region"))
-                .toString();
+    @NonNull
+    @Override
+    public String toString() {
```

Comment:
Im guessing we are planning to use this string for API category user agent which uses ktor and not Kotlin SDK, right?

Reply:
Yes, that and for backward compatibility for APIs which use `UserAgent::string`